### PR TITLE
Fetch API documentation URLs from application readmes

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

During the build, the app now fetches API documentation URLs from application readmes.

✨ New steps:

1. Clones the application repositories referenced in the model.
2. Greps the `README.md` file for an "API docs" badge.
3. If it exists, extracts the API docs URL.

Existing steps:

4. Creates an `exports/docs/apidocs.md` file with all the apps.
5. Publishes the result to [Structurizr](https://structurizr.com/share/56937/documentation#%2F:Published%20APIs) after merges.

**Requirements**

- Only works with publicly cloneable repos
- Picks up [API docs](https://example.org) badges:
    ```
    [![API docs](https://img.shields.io/badge/API_docs-view-85EA2D.svg?logo=swagger)](https://link.to/swagger)
    ```
- More specifically, has to match `\[!\[API docs\(.*\)]\((.*)\)`

## What is the intent behind these changes?

Automatically pick up API documentation from git repos: reduces the manual maintenance burden.

### CODE REVIEW

File: src/main/kotlin/defineDocumentation.kt [hunk 1]
```diff
@@ -44,9 +47,50 @@ private fun writeAPI(w: PrintWriter, apiContainer: Container) {
...
+
+      if (url != null) {
+        if (oldUrl != null) {
+          println("[defineDocumentation] overriding API docs URL\n - from ${oldUrl}\n - to   $url")
+        }
+        APIDocs(url).addTo(it)
+      }
+    }
+
+  return containers.filter { APIDocs.getFrom(it) != null }
```
Comment (mxco86) [inline src/main/kotlin/defineDocumentation.kt:77]: Could we roll these functions into a single chain?

File: src/main/kotlin/pullADRs.kt [hunk 2]
```diff
@@ -2,10 +2,7 @@ package uk.gov.justice.hmpps.architecture
 
 import com.structurizr.Workspace
 import com.structurizr.documentation.AdrToolsImporter
-import com.structurizr.model.SoftwareSystem
-import org.eclipse.jgit.api.Git
 import uk.gov.justice.hmpps.architecture.annotations.ADRSource
-import java.io.File
 
 fun pullADRs(workspace: Workspace) {
   val systemsWithADRs = workspace.model.softwareSystems.stream().filter { ADRSource.getFrom(it) != null }
```
Comment (mxco86) [inline src/main/kotlin/pullADRs.kt:8]: I think this can be collapsed into a single chain

**Review by mxco86 — COMMENTED** [review]
Few minor comments but overall looks good

File: src/main/kotlin/pullADRs.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/pullADRs.kt]: done in 6f21e4af28371704702e0edca372bcdcc13702a1

File: src/main/kotlin/defineDocumentation.kt (hunk 1 above)
Comment (sldblog) [reply src/main/kotlin/defineDocumentation.kt]: done in 6f21e4af28371704702e0edca372bcdcc13702a1

**Review by mxco86 — APPROVED** [review]
👍

**Pull request merged**